data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -172,10 +173,15 @@ public void clearParameters() {
     }
 
     /**
-     * Gets the amount of time a query may execute on the service before it times out. Value must be between 1 minute and 1 hour,
+     * Gets the amount of time a query may execute on the service before it times out. Value must be between 1 second and 1 hour,
      * and so if the value had been set below the minimum or above the maximum, the value returned will be adjusted accordingly.
      */
     public Long getTimeoutInMilliSec() {
+        Object noTimeout = getOption(OPTION_SERVER_TIMEOUT_NO_REQUEST_TIMEOUT);
```

Comment:
This feels wrong.
I don't think this public API should consider the other property just cause we need it somewhere

Reply:
IDK, the api doesn't get the property as-is, it always does a conversion to ms anyway.

So it's not about getting the option itself, but the effective timeout, which is why i think it's corect to have both options considered.

Reply:
Effective timeout sounds like the job that `determineTimeout` is doing.
If this is also this APIs job, name it accordingly